apps/api/src/app/workflows-v2/usecases/placeholder-enrichment/payload-preview-value-generator.usecase.ts

```diff
@@ -22,20 +18,29 @@ export class CreateMockPayloadUseCase {
     }
 
     const controlValue = controlValues[controlValueKey];
-    if (typeof controlValue === 'object') {
-      return this.buildPayloadForEmailEditor(controlValue);
+    const safeAttemptToParseEmailSchema = this.safeAttemptToParseEmailSchema(controlValue);
+    if (safeAttemptToParseEmailSchema) {
+      return safeAttemptToParseEmailSchema;
     }
 
     return this.buildPayloadForRegularText(controlValue);
   }
 
-  private buildPayloadForEmailEditor(controlValue: unknown): Record<string, unknown> {
-    const collectPlaceholderMappings = this.collectPlaceholdersFromTipTapSchemaUsecase.execute({
-      node: controlValue as TipTapNode,
-    });
-    const transformPlaceholderMap = this.transformPlaceholderMapUseCase.execute({ input: collectPlaceholderMappings });
+  private safeAttemptToParseEmailSchema(controlValue: string) {
+    try {
+      const { nestedPayload } = this.hydrateEmailSchemaUseCase.execute({
+        emailEditor: controlValue,
+        masterPayload: {
```

Comment:
just because there will be more, we havn't done actor yet,. that's the reason I added this concept, because its a higher level payload that includes all not only the customer provided payload, I want to minimize the changes when we add actor and others to the framework, create a structure that minimizes the maintenance, 
eventually the placeholders are not reffering to payload, they reffer to such a master payload.